From 502f5a73d5ed4f78327d0a45f5cd32e8c17653b2 Mon Sep 17 00:00:00 2001
From: Wein <kuo.wein@inventec.com>
Date: Thu, 10 Mar 2022 10:03:24 +0800
Subject: [PATCH] gen patch 33

---
 drivers/clk/clk-ast2600.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/clk-ast2600.c b/drivers/clk/clk-ast2600.c
index c5aa74c3a30a..c717c353b127 100644
--- a/drivers/clk/clk-ast2600.c
+++ b/drivers/clk/clk-ast2600.c
@@ -69,7 +69,7 @@
 #define ASPEED_G6_DEF_MAC12_DELAY_1G	0x0041b75d
 #define ASPEED_G6_DEF_MAC12_DELAY_100M	0x00417410
 #define ASPEED_G6_DEF_MAC12_DELAY_10M	0x00417410
-#define ASPEED_G6_DEF_MAC34_DELAY_1G	0x00104208
+#define ASPEED_G6_DEF_MAC34_DELAY_1G 	0x001043C8
 #define ASPEED_G6_DEF_MAC34_DELAY_100M	0x00104208
 #define ASPEED_G6_DEF_MAC34_DELAY_10M	0x00104208
 
@@ -1071,7 +1071,7 @@ static void __init aspeed_g6_cc_init(struct device_node *np)
 		     ASPEED_G6_DEF_MAC34_DELAY_10M);
 
 	/* MAC3/4 default pad driving strength */
-	regmap_write(map, ASPEED_G6_MAC34_DRIVING_CTRL, 0x0000000a);
+	regmap_write(map, ASPEED_G6_MAC34_DRIVING_CTRL, 0x0000000f);
 
 	regmap_read(map, ASPEED_MAC12_CLK_DLY, &reg_1g.w);
 	regmap_read(map, ASPEED_MAC12_CLK_DLY_100M, &reg_100.w);
-- 
2.17.1

